security-analysis/src/test/java/com/powsybl/security/SecurityAnalysisParametersTest.java

```diff
@@ -0,0 +1,96 @@
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.security;
+
+import com.google.auto.service.AutoService;
+import com.powsybl.commons.config.PlatformConfig;
+import com.powsybl.commons.extensions.AbstractExtension;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mockito;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+
+/**
+ * @author Teofil Calin BANC <teofil-calin.banc at rte-france.com>
+ */
+public class SecurityAnalysisParametersTest {
+
+    protected PlatformConfig config;
+
+    @Before
+    public void setUp() {
+        config = Mockito.mock(PlatformConfig.class);
+    }
+
+    @Test
+    public void testExtensions() {
+        SecurityAnalysisParameters parameters = new SecurityAnalysisParameters();
+        DummyExtension dummyExtension = new DummyExtension();
+        parameters.addExtension(DummyExtension.class, dummyExtension);
+
+        assertEquals(1, parameters.getExtensions().size());
+        assertEquals(true, parameters.getExtensions().contains(dummyExtension));
+        assertEquals(true, parameters.getExtensionByName("dummyExtension") instanceof DummyExtension);
+        assertEquals(true, parameters.getExtension(DummyExtension.class) instanceof DummyExtension);
+    }
+
+    @Test
+    public void testNoExtensions() {
+        SecurityAnalysisParameters parameters = new SecurityAnalysisParameters();
+
+        assertEquals(0, parameters.getExtensions().size());
+        assertEquals(false, parameters.getExtensions().contains(new DummyExtension()));
+        assertEquals(false, parameters.getExtensionByName("dummyExtension") instanceof DummyExtension);
+        assertEquals(false, parameters.getExtension(DummyExtension.class) instanceof DummyExtension);
+    }
+
+    @Test
+    public void testExtensionFromConfig() {
+        SecurityAnalysisParameters parameters = SecurityAnalysisParameters.load(config);
+
+        assertEquals(1, parameters.getExtensions().size());
+        assertEquals(true, parameters.getExtensionByName("dummyExtension") instanceof DummyExtension);
+        assertNotNull(parameters.getExtension(DummyExtension.class));
+    }
+
+    private static class DummyExtension extends AbstractExtension<SecurityAnalysisParameters> {
+
+        @Override
+        public String getName() {
+            return "dummyExtension";
+        }
+    }
+
+    @AutoService(ParametersConfigLoader.class)
+    public static class DummyLoader implements ParametersConfigLoader<DummyExtension> {
+
+        public DummyLoader() {
+        }
+
+        @Override
+        public DummyExtension load(PlatformConfig platformConfig) {
+            return new DummyExtension();
+        }
+
+        @Override
+        public String getExtensionName() {
+            return "dummyExtension";
+        }
+
+        @Override
+        public String getCategoryName() {
+            return "dummy-category";
```

Comment:
Should be security-analysis-parameter to avoid mistakes if another dummy extension is added one day